### PR TITLE
Ongoing Project on Active Records

### PR DESCRIPTION
**Pull request by uzorjchibuzor**
@talihomz, we are going to send you the link to the review guidelines when we feel that we have completed the project.  Thank you

### CODE REVIEW

**Comment by uzorjchibuzor** [comment]
[Here](https://docs.google.com/document/d/1glYSqZB6K8wgp-AiqvooG9f3w4W5c2i46q2AoSK00GM/edit) is the link to the review guidelines @talihomz

File: db/migrate/20181018152839_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreatePosts < ActiveRecord::Migration[5.2]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.string :url
+      t.references :user, foreign_key: true
```
Comment (heyapricot) [inline db/migrate/20181018152839_create_posts.rb:6]: There are cases where you don't know from the beginning you need a relation to another model. If that's the case [you can use a migration (and a generator) just to add the reference.](https://edgeguides.rubyonrails.org/active_record_migrations.html#creating-a-migration)

E.g. `rails generate migration AddUserRefToProducts user:references`

Comment (uzorjchibuzor) [reply db/migrate/20181018152839_create_posts.rb:6]: Thanks. I am usually stumped with those kind of things. 

EDIT: @heyalvaro Let me know when to merge

File: test/models/user_test.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  end
+  test "name should be present" do
+    @user.name = ""
+    assert_not @user.valid?
+  end
+   test "name should not be too long" do
+    @user.name = "a"*21
+    assert_not @user.valid?
+   end
+end
```
Comment (heyapricot) [inline test/models/user_test.rb:19]: This one looks good. Only concern is you added a validation for email uniqueness but it is not tested

Comment (heyapricot) [reply test/models/user_test.rb:19]: These are enough validations for this exercise, but for future projects, you could add more validations like testing that the email is a valid string with a regex or making sure that the email is stored as lowercase.

File: test/models/post_test.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  test "title should not be too long" do
+    @post.title = "a"*31
+    assert_not @post.valid?
+  end
+
+  test "url should not be empty" do
+    @post.url = "   "
+    assert_not @post.valid?
+  end
+end
```
Comment (heyapricot) [inline test/models/post_test.rb:39]: Good work. It'd be nice to also test that User.posts.build works (Does it give the post the correct user_id value automatically?)

File: test/models/comment_test.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    @comment.user_id = nil
+    assert_not @comment.valid?
+  end
+
+  test "comment must have a post_id assigned" do
+    @comment.post_id = nil
+    assert_not @comment.valid?
+  end
+
+end
```
Comment (heyapricot) [inline test/models/comment_test.rb:36]: Funny thing, here you tested that the comment has a user_id and a post_id but in the post test you don't check for user_id. Well done

File: test/fixtures/comments.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+one:
+  body: MyText
+  user: one
+  post: one
+
+two:
+  body: MyText
+  user: two
+  post: two
```
Comment (heyapricot) [inline test/fixtures/comments.yml:11]: I'm guessing these .yml files are generate by default, right?

File: db/migrate/20181018152839_create_posts.rb (hunk 1 above)
Comment (heyapricot) [reply db/migrate/20181018152839_create_posts.rb:6]: Everything looks good. Just add a test to check that the post has a user_id (similar to what you did in the test for comments and merge)

File: test/fixtures/comments.yml (hunk 5 above)
Comment (uzorjchibuzor) [reply test/fixtures/comments.yml:11]: yeah.

**Pull request merged**